.github/workflows/ci-test-workflow.yml

```diff
@@ -36,8 +36,7 @@ jobs:
     - name: Setup GitRepository/Kustomization
       run: |
         set -ex
-        if test -n "${GITHUB_HEAD_REF#refs/heads/}"; then flux create source git repo --url=https://github.com/$GITHUB_REPOSITORY.git --branch=${GITHUB_HEAD_REF#refs/heads/}; else flux create source git repo --url=https://github.com/$GITHUB_REPOSITORY.git --branch=${GITHUB_REF#refs/heads/}; fi
-        #flux create source git repo --url=https://github.com/$GITHUB_REPOSITORY.git --branch=$GITHUB_REF_NAME
+        if test -n "${GITHUB_HEAD_REF#refs/heads/}"; then flux create source git repo --url=https://github.com/$GITHUB_REPOSITORY.git --branch=${GITHUB_HEAD_REF#refs/heads/}; else flux create source git repo --url=https://github.com/$GITHUB_REPOSITORY.git --branch=${GITHUB_REF_NAME}; fi
```

Comment:
Hmm, when I did this, GITHUB_REF_NAME would not have pointed to the correct branch for whatever reasons.
If this is fixed now, that's great, but please double-check.

Reply:
I did some testing. Setting Commit hash directly, switching flux git implementation and other things...

Seems like Github automatically creates a Commit+Ref for PR's that is checked out in the Github Action, but is hard to configure for flux. Hence, falling back to previous behavior.